openstates/il/bills.py

```diff
@@ -266,25 +283,169 @@ def scrape(self, session=None):
+                    line = summary_text[x]
+                    if "Short description:" in line:
+                        bill_title = summary_text[x + 1]
+
+                bill = Bill(
+                    bill_id,
+                    legislative_session=session,
+                    title=bill_title,
+                    chamber=chamber,
+                    classification=classification,
+                )
+                bill.add_source(summary_page_url)
+                bill.add_source(url)
+
+                # Sponsors
+                for sponsor in sponsors:
+                    if sponsor.text_content():
+                        bill.add_sponsorship(
+                            name=sponsor.text_content(),
+                            classification="cosponsor",
+                            entity_type="person",
+                            primary=False,
+                        )
 
-        # TODO: get joint session resolution added to python-opencivicdata
-        # for bill_url in self.get_bill_urls(chamber, session_id, 'JSR'):
-        #     bill, votes = self.scrape_bill(chamber, session_id, 'JSR', bill_url,
-        #                                    'joint session resolution')
-        #     yield bill
-        #     yield from votes
+                # Bill version
+                version_url = bill_doc.xpath('//a[contains (., "Full Text")]/@href')[0]
+                bill.add_version_link(bill_id, version_url, media_type="text/html")
+
+                # Actions
+                bill_text = bill_doc.xpath("//pre")
+                if bill_text:
+                    bill_text = bill_text[0].text_content().splitlines()
+                    for x in range(len(bill_text)):
+                        line = bill_text[x].split()
+                        # Regex is looking for this format: JAN-11-2001 or 99-02-17
+                        if line and (
+                            re.match(r"\D\D\D-\d\d-\d\d\d\d", line[0])
+                            or re.match(r"\d\d-\d\d-\d\d", line[0])
+                        ):
+                            if session in ["91st", "90th"]:
+                                action_date = datetime.datetime.strptime(
+                                    line[0], "%y-%m-%d"
+                                )
+                            else:
+                                action_date = datetime.datetime.strptime(
+                                    line[0], "%b-%d-%Y"
+                                )
+
+                            action_date = eastern.localize(action_date)
+                            action_date = action_date.isoformat()
+
+                            action = " ".join(line[2:])
+                            if line[1] == "S":
+                                action_chamber = "upper"
+                            else:
+                                action_chamber = "lower"
+
+                            for pattern, atype in _archived_action_classifiers.items():
```

Comment:
wouldn't this loop set every action to the last atype if there isn't a break?

look at using an else clause on for https://book.pythontips.com/en/latest/for_-_else.html

Reply:
Fixed!